app/database/admin/data_operator/index.ts

```diff
@@ -1,183 +1,703 @@
-     * the 'SERVER' database.
-     * @param {string} tableName
-     * @returns {Promise<DBInstance>}
-     */
-    private getDatabase = async (tableName: string): Promise<DBInstance> => {
-        const isInDefaultDB = Object.values(MM_TABLES.DEFAULT).some((tbName) => {
-            return tableName === tbName;
-        });
-
-        if (isInDefaultDB) {
-            return this.defaultDatabase || this.getDefaultDatabase();
-        }
-
-        return this.serverDatabase || this.getServerDatabase();
-    };
-
-    /**
-     * getDefaultDatabase: Returns the default database
-     * @returns {Promise<DBInstance>}
-     */
-    private getDefaultDatabase = async () => {
-        this.defaultDatabase = await DatabaseManager.getDefaultDatabase();
-        return this.defaultDatabase;
-    };
-
-    /**
-     * getServerDatabase: Returns the current active server database (multi-server support)
-     * @returns {Promise<DBInstance>}
-     */
-    private getServerDatabase = async () => {
-        this.serverDatabase = await DatabaseManager.getActiveServerDatabase();
-        return this.serverDatabase;
-    };
+  /**
+   * handleIsolatedEntity: Handler responsible for the Create/Update operations on the isolated entities as described
+   * by the IsolatedEntities enum
+   * @param {HandleIsolatedEntityData} entityData
+   * @param {IsolatedEntities} entityData.tableName
+   * @param {Records} entityData.values
+   * @returns {Promise<void>}
+   */
+  handleIsolatedEntity = async ({tableName, values}: HandleIsolatedEntityData) => {
+      let recordOperator;
+
+      switch (tableName) {
+          case IsolatedEntities.APP: {
+              recordOperator = operateAppRecord;
+              break;
+          }
+          case IsolatedEntities.GLOBAL: {
+              recordOperator = operateGlobalRecord;
+              break;
+          }
+          case IsolatedEntities.SERVERS: {
+              recordOperator = operateServersRecord;
+              break;
+          }
+          case IsolatedEntities.CUSTOM_EMOJI: {
+              recordOperator = operateCustomEmojiRecord;
+              break;
+          }
+          case IsolatedEntities.ROLE: {
+              recordOperator = operateRoleRecord;
+              break;
+          }
+          case IsolatedEntities.SYSTEM: {
+              recordOperator = operateSystemRecord;
+              break;
+          }
+          case IsolatedEntities.TERMS_OF_SERVICE: {
+              recordOperator = operateTermsOfServiceRecord;
+              break;
+          }
+          default: {
+              recordOperator = null;
+              break;
+          }
+      }
+
+      if (recordOperator) {
+          await this.handleBase({values, tableName, recordOperator});
+      }
+  };
+
+  /**
+   * handleDraft: Handler responsible for the Create/Update operations occurring the Draft entity from the 'Server' schema
+   * @param {RawDraft[]} drafts
+   * @returns {Promise<any[]>}
```

Comment:
why is this returning an empty array at all times? is a return needed? if so, what should be returned here?

Also why is it `any` in the return array type?

Reply:
same applies to the other handlers

Reply:
We have two ESLINT rules acting in this situation.  The first one is 'consistent return' and the other one being 'all code path should return or no implicit return' from Typescript.

I will review all JSDoc and update the return statements where possible.

For example, in the handlePosts method, we are calling other handlers and concatenating their results into the batch array.  If one of those handlers would return an empty array, that will work fine as we are only concatenating arrays together and not `undefined`.  The alternative would be to verify if the handler does not return `undefined` and if so then we concatenate.  I prefer it returning an empty array in this case.

Reply:
I was talking about the type in the array not the array being empty